src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/KerrSchild.cpp

```diff
@@ -0,0 +1,248 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/KerrSchild.hpp"
+
+#include <numeric>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/GrTags.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace EinsteinSolutions {
+
+KerrSchild::KerrSchild(const double mass, KerrSchild::Spin::type spin,
+                       KerrSchild::Center::type center,
+                       const OptionContext& context)
+    : mass_(mass),
+      // clang-tidy: do not std::move trivial types.
+      spin_(std::move(spin)),  // NOLINT
+      // clang-tidy: do not std::move trivial types.
+      center_(std::move(center))  // NOLINT
+{
+  const double spin_magnitude = magnitude(spin_);
+  if (spin_magnitude > 1.0) {
+    PARSE_ERROR(context,
+                "Spin magnitude must be < 1. Given spin: "
+                    << spin_ << " with magnitude " << spin_magnitude);
+  }
+  if (mass_ < 0.0) {
+    PARSE_ERROR(context, "Mass must be non-negative. Given mass: " << mass_);
+  }
+}
+
+void KerrSchild::pup(PUP::er& p) noexcept {
+  p | mass_;
+  p | spin_;
+  p | center_;
+}
+
+template <typename DataType>
+tuples::TaggedTupleTypelist<KerrSchild::tags<DataType>> KerrSchild::solution(
+    const tnsr::I<DataType, 3>& x, const double /*t*/) const noexcept {
+  const auto a_squared =
+      std::inner_product(spin_.begin(), spin_.end(), spin_.begin(), 0.);
+  //-:  const auto a_squared = [](const auto& spin) {
+  //-:    double l_a_squared = 0.0;
+  //-:    for (const auto& s : spin) {
+  //-:      l_a_squared += square(s);
+  //-:    }
+  //-:    return l_a_squared;
+  //-:  }(spin_);
+
+  const auto x_minus_center = [](const auto& coord, const auto& center) {
```

Comment:
replace auto in lambda with actual types. The lambda does not need to be a template function

mark noexcept